### PR TITLE
[Network] Support vpn gateway generation

### PR DESCRIPTION
**Pull request by jsntcy**
**Description of PR (Mandatory)**  
This PR will support vpn gateway generation which has detailed information in #11772.

**Testing Guide**  
- Run command `az network vnet-gateway create -g {rg} -n {gw1} --vnet {vnet1_id} --public-ip-address {ip1} --vpn-gateway-generation Generation1` and check if the command response has vpnGatewayGeneration with `Generation1`

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 1]
```diff
@@ -3016,11 +3020,12 @@ def test_network_vpn_gateway(self, resource_group):
             self.check('sku.capacity', 2),
             self.check('sku.name', 'Basic'),
             self.check('vpnType', 'RouteBased'),
+            self.check('vpnGatewayGeneration', 'Generation1'),
             self.check('enableBgp', False),
             self.check('customRoutes.addressPrefixes[0]', self.kwargs['custom_routes1'])
         ])
 
-        self.cmd('network vnet-gateway update -g {rg} -n {gw1} --custom-routes {custom_routes1} {custom_routes2}', checks=[
+        self.cmd('network vnet-gateway update -g {rg} -n {gw1} --sku Standard --custom-routes {custom_routes1} {custom_routes2}', checks=[
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: Fix existing live test error (not related to this PR)

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 2]
```diff
@@ -3003,7 +3003,11 @@ def test_network_vpn_gateway(self, resource_group):
             'vnet2_id': '/subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Network/virtualNetworks/{vnet2}'.format(**self.kwargs)
         })
 
-        self.cmd('network vnet-gateway create -g {rg} -n {gw1} --vnet {vnet1_id} --public-ip-address {ip1} --custom-routes {custom_routes1} --no-wait')
+        with self.assertRaisesRegexp(CLIError, 'vpn_gateway_generation should not be provided if gateway_type is not Vpn.'):
+            self.cmd(
+                'network vnet-gateway create -g {rg} -n {gw1} --vnet {vnet1_id} --public-ip-address {ip1} --gateway-type ExpressRoute --vpn-gateway-generation Generation1')
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: Is it going to raise error by design?

Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: yes

---
In reply to: [397627241](https://github.com/Azure/azure-cli/pull/12740#discussion_r397627241) [](ancestors = 397627241)

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -5212,7 +5212,7 @@ def _prep_cert_create(cmd, gateway_name, resource_group_name):
 
 def create_vnet_gateway(cmd, resource_group_name, virtual_network_gateway_name, public_ip_address,
```
[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: Let me check

---
In reply to: [397624073](https://github.com/Azure/azure-cli/pull/12740#discussion_r397624073) [](ancestors = 397624073)

[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: add support for "update" cmd

---
In reply to: [397650578](https://github.com/Azure/azure-cli/pull/12740#discussion_r397650578) [](ancestors = 397650578,397624073)

[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: No, vpn_gateway_generation cannot be updated.

---
In reply to: [397650578](https://github.com/Azure/azure-cli/pull/12740#discussion_r397650578) [](ancestors = 397650578,397624073)

**Pull request merged**